¡Hola!

Mi nombre es Marcos Torres y tengo el gusto de revisar tu código el día de hoy.

Cuando vea algo notable o algún asunto en el notebook, te dejaré un comentario o un hint. Se que encontraras la mejor respuesta para resolver todos los comentarios, de no ser así, no te preocupes en futuras iteraciones dejaré comentarios y pistas más específicos.

Este proceso es muy parecido al que se recibe de un gerente o de un Senior Data Scientist en un trabajo real, por lo que te estarás preparando para la experiencia en la vida real.

Encontrarás comentarios en verde, amarillo o rojo como los siguientes:

<div class="alert alert-block alert-success">
<b>Comentario del revisor</b> <a class="tocSkip"></a>

Buen trabajo. ¡Lo hiciste muy bien!
</div>

<div class="alert alert-block alert-warning">
<b>Comentario del revisor</b> <a class="tocSkip"></a>

Nota. Se puede mejorar.
</div>

<div class="alert alert-block alert-danger">
<b>Comentario del revisor</b> <a class="tocSkip"></a>

Necesitas corregirlo. Este bloque indica que se requiere una correción. El trabajo no se acepta si tiene estos bloques.
</div>

Puedes responder a mis comentarios usando estos bloques:

<div class="alert alert-block alert-info">
<b>Respuesta del estudiante.</b> <a class="tocSkip"></a>
</div>

## Descarga y prepara los datos. Explica el procedimiento.

In [1]:
# Cargar todas las librerías
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error
import matplotlib.pyplot as plt
from scipy import stats

In [2]:
#Abre y examina el archivo de datos
region_1 = pd.read_csv('https://practicum-content.s3.us-west-1.amazonaws.com/datasets/geo_data_0.csv') 
region_2 = pd.read_csv('https://practicum-content.s3.us-west-1.amazonaws.com/datasets/geo_data_1.csv') 
region_3 = pd.read_csv('https://practicum-content.s3.us-west-1.amazonaws.com/datasets/geo_data_2.csv') 

<div class="alert alert-block alert-success">
<b>Comentario del revisor</b> <a class="tocSkip"></a>

Bien, usaste una celda independiente para importar las librerías y otra para leer los datos.
</div>

In [3]:
region_1.info()
region_1.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   id       100000 non-null  object 
 1   f0       100000 non-null  float64
 2   f1       100000 non-null  float64
 3   f2       100000 non-null  float64
 4   product  100000 non-null  float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB


f0             f1             f2        product
count  100000.000000  100000.000000  100000.000000  100000.000000
mean        0.500419       0.250143       2.502647      92.500000
std         0.871832       0.504433       3.248248      44.288691
min        -1.408605      -0.848218     -12.088328       0.000000
25%        -0.072580      -0.200881       0.287748      56.497507
50%         0.502360       0.250252       2.515969      91.849972
75%         1.073581       0.700646       4.715088     128.564089
max         2.362331       1.343769      16.003790     185.364347

In [4]:
region_1.sample(5)

id        f0        f1        f2     product
24445  iBNbU -0.410031  0.833765  3.133170   87.527460
7048   ZNJMF  0.115549  1.160076  3.133662   54.386876
37776  uYpqQ  0.007175  0.810077  0.700358   48.515852
96735  ZuBWI  1.665934 -0.366112  6.161269  159.512977
28631  omRax -0.377964  0.928260  4.426847  114.576698

In [5]:
region_1.duplicated().sum()

0

In [6]:
region_1.isna().sum()

id         0
f0         0
f1         0
f2         0
product    0
dtype: int64

<div class="alert alert-block alert-success">
<b>Comentario del revisor</b> <a class="tocSkip"></a>

Bien, se exploraron correctamente los valores faltantes.
</div>

In [7]:
region_2.info()
region_2.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   id       100000 non-null  object 
 1   f0       100000 non-null  float64
 2   f1       100000 non-null  float64
 3   f2       100000 non-null  float64
 4   product  100000 non-null  float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB


f0             f1             f2        product
count  100000.000000  100000.000000  100000.000000  100000.000000
mean        1.141296      -4.796579       2.494541      68.825000
std         8.965932       5.119872       1.703572      45.944423
min       -31.609576     -26.358598      -0.018144       0.000000
25%        -6.298551      -8.267985       1.000021      26.953261
50%         1.153055      -4.813172       2.011479      57.085625
75%         8.621015      -1.332816       3.999904     107.813044
max        29.421755      18.734063       5.019721     137.945408

In [8]:
region_2.sample(5)

id         f0        f1        f2    product
59252  9dhVi  13.650371  6.181164  3.000312  80.859783
89547  qAQUB   1.849943  2.471807  0.997354  26.953261
80414  CK77O  -5.514299 -7.614751  2.003262  57.085625
2990   Y6Yyp  -4.383352 -9.663722 -0.000487   3.179103
59017  RpFQ8  16.200382  4.227939  1.000596  26.953261

In [9]:
region_2.duplicated().sum()

0

In [10]:
region_2.isna().sum()

id         0
f0         0
f1         0
f2         0
product    0
dtype: int64

In [11]:
region_3.info()
region_3.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 5 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   id       100000 non-null  object 
 1   f0       100000 non-null  float64
 2   f1       100000 non-null  float64
 3   f2       100000 non-null  float64
 4   product  100000 non-null  float64
dtypes: float64(4), object(1)
memory usage: 3.8+ MB


f0             f1             f2        product
count  100000.000000  100000.000000  100000.000000  100000.000000
mean        0.002023      -0.002081       2.495128      95.000000
std         1.732045       1.730417       3.473445      44.749921
min        -8.760004      -7.084020     -11.970335       0.000000
25%        -1.162288      -1.174820       0.130359      59.450441
50%         0.009424      -0.009482       2.484236      94.925613
75%         1.158535       1.163678       4.858794     130.595027
max         7.238262       7.844801      16.739402     190.029838

In [12]:
region_3.sample(5)

id        f0        f1        f2     product
89593  w5qlX -0.109187  0.187493  1.217335   63.885056
37091  0DhJD -2.095987  2.158162  6.440236  115.862158
12697  FZ8q6 -0.950188  1.721238  8.246901  148.121842
29359  ot0SW -2.739368 -1.055905  4.971578  178.733284
48466  ZCSLN  3.645640  3.059847  0.946516   60.258458

<div class="alert alert-block alert-success">
<b>Comentario del revisor</b> <a class="tocSkip"></a>

Buen uso de los métodos de pandas para explorar los datos.
</div>

In [13]:
region_3.duplicated().sum()

0

In [14]:
region_3.isna().sum()

id         0
f0         0
f1         0
f2         0
product    0
dtype: int64

Iniciamos el proyecto importando los datos con los cuales vamos a trabajar. Hallamos la información y descripción de las tablas, buscamos si hay algunos datos ausentes o duplicados. Y vemos que todo luce bien, no hay la necesidad de hacer algunos cambios en los datos.

## Entrena y prueba el modelo para cada región:

### Región 1

In [15]:
# Divide los datos en un conjunto de entrenamiento y un conjunto de validación en una proporción de 75:25
features = region_1.drop(['id', 'product'], axis=1)
target = region_1['product']
features_train, features_val, target_train, target_val = train_test_split(features, target, test_size=0.25, random_state=12345)
print(features_train.shape)
print(features_val.shape)

# Entrena el modelo y haz predicciones para el conjunto de validación.
model_1 = LinearRegression()
model_1.fit(features_train, target_train)
predicted_valid = model_1.predict(features_val)

# Guarda las predicciones y las respuestas correctas para el conjunto de validación.
predictions_1 = pd.Series(predicted_valid, index=target_val.index)
correct_answers_1 = target_val

# Muestra el volumen medio de reservas predicho y RMSE del modelo.
rmse = mean_squared_error(correct_answers_1,predictions_1, squared=False)
desviacion_estandar = np.std(correct_answers_1, ddof=1) 
print(f"Desviación Estándar real: {desviacion_estandar}")
print("Media Real:", correct_answers_1.mean())
print("Media Predicha:", predictions_1.mean())
print("RMSE:", rmse)

(75000, 3)
(25000, 3)
Desviación Estándar real: 44.286912625663646
Media Real: 92.07859674082927
Media Predicha: 92.59256778438035
RMSE: 37.5794217150813


<div class="alert alert-block alert-success">
<b>Comentario del revisor</b> <a class="tocSkip"></a>

Se entrenó correctamente el modelo para esta región.
</div>

Análisis de los resultados de la Región 1:

- El promedio de los valores reales es de 92.08 y el promedio de los valores que el modelo ha predicho es de 92.59.

- La dispersión de los valores reales alrededor de su media es de 44.29. 

- La diferencia promedio entre los valores reales y los valores predichos es de aproximadamente 37.58 unidades. 

La media real y la media predicha son bastante similares, lo que indica que el modelo tiene un buen rendimiento en términos de predicciones promedio.

La desviación estándar real nos dice cuánta variabilidad hay en los datos. Esta desviación estándar es alta, por lo que hay muchos valores que pueden alejarse significativamente de la media.

En este caso, el RMSE (37.58) es menor que la desviación estándar real (44.29). Esto sugiere que, aunque hay una buena cantidad de variabilidad en los datos, el modelo está haciendo un trabajo razonable al predecir, ya que las desviaciones de las predicciones no son extremadamente grandes en comparación con la dispersión total de los datos.

### Región 2

In [16]:
# Divide los datos en un conjunto de entrenamiento y un conjunto de validación en una proporción de 75:25

features = region_2.drop(['id', 'product'], axis=1)
target = region_2['product']
features_train, features_val, target_train, target_val = train_test_split(features, target, test_size=0.25, random_state=12345)
print(features_train.shape)
print(features_val.shape)

# Entrena el modelo y haz predicciones para el conjunto de validación.
model_2 = LinearRegression()
model_2.fit(features_train, target_train)
predicted_valid = model_2.predict(features_val)

# Guarda las predicciones y las respuestas correctas para el conjunto de validación.
predictions_2 = pd.Series(predicted_valid, index=target_val.index)
correct_answers_2= target_val

# Muestra el volumen medio de reservas predicho y RMSE del modelo.
rmse = mean_squared_error(correct_answers_2,predictions_2, squared=False)
desviacion_estandar = np.std(correct_answers_2, ddof=1) 
print(f"Desviación Estándar real: {desviacion_estandar}")
print("Media Real:", correct_answers_2.mean())
print("Media Predicha:", predictions_2.mean())
print("RMSE:", rmse)

(75000, 3)
(25000, 3)
Desviación Estándar real: 46.02216537514022
Media Real: 68.72313602435997
Media Predicha: 68.728546895446
RMSE: 0.893099286775617


<div class="alert alert-block alert-success">
<b>Comentario del revisor</b> <a class="tocSkip"></a>

Muy bien, las métricas son acorde a lo esperado para la región 2.
</div>

Análisis de los resultados de la Región 2:

- La media aritmética de los valores reales es aproximadamente 68.72.

- La media predicha de 68.73 es el promedio de los valores que tu modelo ha predicho.

- Esta desviación estándar de 46.02 sugiere que hay una amplia dispersión en los valores reales, lo que significa que los resultados están bastante separados de la media.

- Este RMSE de 0.89 indica que, en promedio, las predicciones del modelo se desvían de los valores reales en solo 0.89 unidades. Esto sugiere que el modelo está haciendo un buen trabajo en la predicción.

La media predicha es casi idéntica a la media real, lo que indica que el modelo tiene un buen ajuste en términos de nivel promedio.

El RMSE es muy bajo en comparación con la desviación estándar real. A pesar de que los datos son bastante dispersos (como indica la desviación estándar), las predicciones del modelo son muy precisas. Por lo tanto, el modelo logra predecir los valores con muy poca variabilidad en comparación con la variabilidad total de los datos.

En resumen, este modelo de regresión lineal parece funcionar muy bien: las predicciones son muy cercanas a los valores reales, y el RMSE bajo indica una buena precisión. Esto sugiere que, aunque hay una gran variabilidad en los datos, y el modelo logra capturar la tendencia general de manera efectiva.

### Región 3

In [17]:
# Divide los datos en un conjunto de entrenamiento y un conjunto de validación en una proporción de 75:25

features = region_3.drop(['id', 'product'], axis=1)
target = region_3['product']
features_train, features_val, target_train, target_val = train_test_split(features, target, test_size=0.25, random_state=12345)
print(features_train.shape)
print(features_val.shape)

# Entrena el modelo y haz predicciones para el conjunto de validación.
model_3 = LinearRegression()
model_3.fit(features_train, target_train)
predicted_valid = model_3.predict(features_val)

# Guarda las predicciones y las respuestas correctas para el conjunto de validación.
predictions_3 = pd.Series(predicted_valid, index=target_val.index)
correct_answers_3 = target_val

# Muestra el volumen medio de reservas predicho y RMSE del modelo.
rmse = mean_squared_error(correct_answers_3,predictions_3, squared=False)
desviacion_estandar = np.std(correct_answers_3, ddof=1) 
print(f"Desviación Estándar real: {desviacion_estandar}")
print("MEDIA REAL:", correct_answers_3.mean())
print("MEDIA PREDICHA:", predictions_3.mean())
print("RMSE:", rmse)

(75000, 3)
(25000, 3)
Desviación Estándar real: 44.90298244552798
MEDIA REAL: 94.88423280885438
MEDIA PREDICHA: 94.96504596800489
RMSE: 40.02970873393434


Análisis de los resultados de la Región 3:

- El promedio de los valores reales es de 94.88.

- La media de las predicciones es de 94.97. 

- Esta desviación estándar de 44.90 sugiere que los valores están bastante dispersos alrededor de la media.

- Las predicciones del modelo se desvían de los valores reales en aproximadamente 40.03 unidades.

La media predicha es muy similar a la media real, lo que es un buen indicador de que el modelo está capturando adecuadamente la tendencia general de los datos. 

El RMSE de 40.03 es significativo, especialmente considerando la desviación estándar real de 44.90. Esto sugiere que, aunque las predicciones son cercanas al promedio, existe una alta variabilidad en las predicciones individuales. En otras palabras, las predicciones del modelo pueden estar bastante lejos de los valores reales para muchas observaciones.

<div class="alert alert-block alert-success">
<b>Comentario del revisor</b> <a class="tocSkip"></a>

Buenos comentarios sobre la región 3.
</div>

## Prepárate para el cálculo de ganancias:

Almacena todos los valores necesarios para los cálculos en variables separadas.
Dada la inversión de 100 millones por 200 pozos petrolíferos, de media un pozo petrolífero debe producir al menos un valor de 500,000 dólares en unidades para evitar pérdidas (esto es equivalente a 111.1 unidades). Compara esta cantidad con la cantidad media de reservas en cada región.
Presenta conclusiones sobre cómo preparar el paso para calcular el beneficio.

In [18]:
# Definición de variables

inversion_total = 100_000_000  # inversión total en dólares
numero_de_pozos = 200  # número de pozos seleccionados
produccion_media_necesaria = 500_000  # producción media necesaria por pozo en dólares
unidades_por_pozo = 111.1

In [19]:
# Compara 111.1 unidades con la cantidad media de reservas en cada región

data = {
    'Region 1': [predictions_1.mean()],
    'Region 2': [predictions_2.mean()],
    'Region 3': [predictions_3.mean()],
    'Region ideal': [111.1],
}

regiones_reservas = pd.DataFrame(data)
regiones_reservas

Region 1   Region 2   Region 3  Region ideal
0  92.592568  68.728547  94.965046         111.1

<div class="alert alert-block alert-success">
<b>Comentario del revisor</b> <a class="tocSkip"></a>

Bien, me gustó que presentaras la comparación en un data frame para facilitar la comparación.
</div>

Al comparar las medias de reservas en cada región con la media ideal de 111.1 unidades, vemos que las tres regiones están por debajo de este valor, siendo la región 2 la peor opción. La región 1 y 3 presentan medias similares, además de estar más cerca de la idónea. Acá estamos haciendo una observación general de todos los pozos, por lo que es necesario analizar más para cumplir con las condiciones que nos dan posteriormente.

<div class="alert alert-block alert-success">
<b>Comentario del revisor</b> <a class="tocSkip"></a>

Con estos resultados se puede ver porque se busca solo los mejores pozos.
</div>

In [20]:
# cómo preparar el paso para calcular el beneficio.

ingreso_por_barril = 4.5  # ingresos por barril en dólares
ingreso_por_unidad = 4500 # el volumen de reservas está expresado en miles de barriles
top_pozos_region_1 = predictions_1.nlargest(200)
inversion_total = 100_000_000 # presupuesto para el desarrollo de 200 pozos petroleros

# Formula para hallar el beneficio de cada pozo por region

ingresos_region = (top_pozos_region_1 * ingreso_por_barril * 1000) - inversion_total

Para calcular el beneficio por región, primero debemos hallar los 200 pozos con mayores reservas de cada región según lo predecido por los modelos de regresión lineal elaborados,de preferencia estas reservas deberían ser mayor a 111.1 unidades, así como generar un beneficio mínimo de 500,000 dólares para ser considerado rentable y evitar pérdidas.

Se sabe que una unidad genera 4500 dólares, por lo tanto para hallar el ingreso de cada pozo, solo multiplicaríamos el volumen de cada reserva (los 200 pozos con unidades mayores a 111.1) * el precio de cada barril (4.5) * 1000 
,o  multiplicado con el precio por unidad (4500.00), que es lo mismo, solo expresado en diferentes unidades.

Para hallar los ingresos totales por región, es necesario restar el presupuesto para el desarrollo de los 200 pozos petroleros seleecionados, lo cual son 100 millones de dólares.

## Escribe una función para calcular la ganancia de un conjunto de pozos de petróleo seleccionados y modela las predicciones:

Elige los 200 pozos con los valores de predicción más altos de cada una de las 3 regiones (es decir, archivos 'csv').
Resume el volumen objetivo de reservas según dichas predicciones. Almacena las predicciones para los 200 pozos para cada una de las 3 regiones.
Calcula la ganancia potencial de los 200 pozos principales por región. Presenta tus conclusiones: propón una región para el desarrollo de pozos petrolíferos y justifica tu elección.

- El presupuesto para el desarrollo de 200 pozos petroleros es de 100 millones de dólares.
- Un barril de materias primas genera 4.5 USD de ingresos. El ingreso de una unidad de producto es de 4500 dólares (el volumen de reservas está expresado en miles de barriles).

In [21]:
def beneficio(predictions_df):
    top_200 = predictions_df.nlargest(n=200)
    suma_region = top_200.sum() 
    beneficio_total = (suma_region * 4500) - 100000000
    return beneficio_total  

In [22]:
top_region1 = beneficio(predictions_1)
top_region1

39960488.77465129

In [23]:
top_region2 = beneficio(predictions_2)
top_region2

24857120.51973541

In [24]:
top_region3 = beneficio(predictions_3)
top_region3

33217543.962432563

<div class="alert alert-block alert-success">
<b>Comentario del revisor</b> <a class="tocSkip"></a>

La función es correcta.
</div>

La región 1 tiene la suma más alta con 40 millones de dólares aproximadamente, lo que sugiere que esta región podría tener los pozos más productivos en términos de valor.
La región 3 sigue de cerca con 33 millones de dólares, lo que también indica una buena capacidad productiva.
La región 2 tiene la suma más baja con 24 millones de dólares lo que podría indicar que sus pozos son menos productivos o que hay menos pozos de gran valor.

La región 1, al tener la mayor suma, podría ser vista como la más prometedora en términos de desarrollo de recursos. Esto podría implicar que la inversión en esta región podría ofrecer un mejor retorno.

## Calcula riesgos y ganancias para cada región:

Utilizando las predicciones que almacenaste en el paso 4.2, emplea la técnica del bootstrapping con 1000 muestras para hallar la distribución de los beneficios.
Encuentra el beneficio promedio, el intervalo de confianza del 95% y el riesgo de pérdidas. La pérdida es una ganancia negativa, calcúlala como una probabilidad y luego exprésala como un porcentaje.
Presenta tus conclusiones: propón una región para el desarrollo de pozos petrolíferos y justifica tu elección. ¿Coincide tu elección con la elección anterior en el punto 4.3?

- Al explorar la región, se lleva a cabo un estudio de 500 puntos con la selección de los mejores 200 puntos para el cálculo del beneficio.

- Después de la evaluación de riesgo, mantén solo las regiones con riesgo de pérdidas inferior al 2.5%. De las que se ajustan a los criterios, se debe seleccionar la región con el beneficio promedio más alto.

In [25]:
state = np.random.RandomState(12345)

def bootstrapping(beneficios,n_iter=1000):
    bootstrapped_total = []
    for i in range(n_iter):
        pozos_select = beneficios.sample(n=500, replace=True, random_state=state)
        top_200_500 = beneficio(pozos_select)
        bootstrapped_total.append(top_200_500)
    bootstrapped_total = pd.Series(bootstrapped_total)
    return bootstrapped_total

<div class="alert alert-block alert-success">
<b>Comentario del revisor</b> <a class="tocSkip"></a>

¡Excelente! El boostrap se itera correctamente y se obtienen los mejores 200 pozos de los 500 de la submuestra de cada iteración, buen trabajo.
</div>

In [26]:
# beneficio total de la region a partir de los 200 mejores por muestra escogidas de 500 pozos con las 1000 iteraciones
beneficio_region1 = bootstrapping(predictions_1)
beneficio_region2 = bootstrapping(predictions_2)
beneficio_region3 = bootstrapping(predictions_3)

In [27]:
def calcular_metricas(beneficios_muestra):
      
    # Calcular beneficio promedio
    beneficio_promedio = np.mean(beneficios_muestra)
    
    # Calcular intervalo de confianza del 95%
    intervalo_confianza = np.percentile(beneficios_muestra, [2.5, 97.5])
    
    # Calcular riesgo de pérdidas
    riesgo_perdidas = np.mean(beneficios_muestra < 0) * 100  # Porcentaje de muestras con beneficios negativos
    
    return beneficio_promedio, intervalo_confianza, riesgo_perdidas

In [28]:
bpromedio_1, intervalo_confianza_1, riesgo_perdidas_1 = calcular_metricas(beneficio_region1)

print(f"Beneficio Promedio: {bpromedio_1:.2f}")
print(f"Intervalo de Confianza (95%): [{intervalo_confianza_1[0]:.2f}, {intervalo_confianza_1[1]:.2f}]")
print(f"Riesgo de Pérdidas: {riesgo_perdidas_1:.2f}%")

Beneficio Promedio: 3580260.54
Intervalo de Confianza (95%): [1401373.36, 5988772.71]
Riesgo de Pérdidas: 0.10%


In [29]:
bpromedio_2, intervalo_confianza_2, riesgo_perdidas_2 = calcular_metricas(beneficio_region2)

print(f"Beneficio Promedio: {bpromedio_2:.2f}")
print(f"Intervalo de Confianza (95%): [{intervalo_confianza_2[0]:.2f}, {intervalo_confianza_2[1]:.2f}]")
print(f"Riesgo de Pérdidas: {riesgo_perdidas_2:.2f}%")

Beneficio Promedio: 4591672.84
Intervalo de Confianza (95%): [760478.23, 8616321.30]
Riesgo de Pérdidas: 0.80%


In [30]:
bpromedio_3, intervalo_confianza_3, riesgo_perdidas_3 = calcular_metricas(beneficio_region3)

print(f"Beneficio Promedio: {bpromedio_3:.2f}")
print(f"Intervalo de Confianza (95%): [{intervalo_confianza_3[0]:.2f}, {intervalo_confianza_3[1]:.2f}]")
print(f"Riesgo de Pérdidas: {riesgo_perdidas_3:.2f}%")

Beneficio Promedio: 2717422.04
Intervalo de Confianza (95%): [911157.07, 4522088.83]
Riesgo de Pérdidas: 0.10%


<div class="alert alert-block alert-success">
<b>Comentario del revisor</b> <a class="tocSkip"></a>

Buen resumen usando los prints para cada una de las regiones.
</div>

 - Presenta tus conclusiones: propón una región para el desarrollo de pozos petrolíferos y justifica tu elección. ¿Coincide tu elección con la elección anterior en el punto 4.3?
 
- Condiciones del proyecto: después de la evaluación de riesgo, mantén solo las regiones con riesgo de pérdidas inferior al 2.5%. De las que se ajustan a los criterios, se debe seleccionar la región con el beneficio promedio más alto.

De acuerdo a la evaluación de rieso, las tres regiones cumplen la condición. El beneficio promedio más alto lo tiene la región 2, siguiendolo la región 1 y finalmente la región 3. Si sólo nos enfocamos en lo mencionado anteriormente la región 2 sería la elegida, sin embargo elijo la región 1 por las siguientes razones:

- El riesgo de pérdida de la región 1 (0.1%) es mucho menor que la región 2 (0.8%).
- Es cierto que la región 2 tiene un beneficio promedio mayor a la región 1, sim embargo deberíamos observar cual es el rango del intervalo para ambos beneficio promedio. La distancia del promedio con los rangos del intervalo de la región 1 es mucho menor a la de la región 2, por lo tanto hay menos variabilidad en la distribución de los datos de las muestras aleatorias.

Por lo tanto la región 1, es la mejor opción que cumple con las condiciones del proyectos para obtener la mejor rentabilidad.

<div class="alert alert-block alert-success">
<b>Comentario del revisor</b> <a class="tocSkip"></a>

¡Hola! 
    
Te quiero felicitar por tu trabajo, está muy completo y bien realizado. Me gustó que el bootstrap tiene la lógica correcta en la iteración y obtuviste adecuadamente las probabilidades de riesgo de pérdida. Bien, puedo aprobar tu trabajo, éxito en los siguientes sprints.
    
Saludos, Marcos.
</div>